Remove OSTATES from StatesTrajectory doxygen

chrisdembia opened the pull request:
This PR updates the StatesTrajectory documentation to no longer describe the OSTATES file format, as this is currently off the list for 4.0. We can easily add this documentation back as soon as OSTATES file support is in place.

This PR also addresses #1381 .

@tkuchida, would you be interested in reviewing this?

tkuchida commented on OpenSim/Simulation/StatesTrajectory.h:
```diff
@@ -44,20 +44,22 @@ class Model;
  * This class was introduced in OpenSim version 4.0, and enables scripting
  * (Python/MATLAB) and C++ users to postprocess their results with greater ease
- * and more versatility than with an Analysis.
+ * and more flexibly than with an Analysis.
```
"with greater ease and flexibility"?

tkuchida commented on OpenSim/Simulation/StatesTrajectory.h:
```diff
@@ -44,20 +44,22 @@ class Model;
- * @note This class is a work in progress. The ability to read and write a
- * StatesTrajectory to an OSTATES is not available yet, even though the
- * documentation is written as if this ability exists.
+ * @note In a future release, we hope to support an OSTATES file format that
```
"hope" -> "plan"

chrisdembia commented:
@tkuchida I have addressed your comments. 



> Built doxygen locally.

Ah! I could have omitted `[ci skip]` for the initial PR/commit. Sorry about that.

tkuchida commented on OpenSim/Simulation/StatesTrajectory.h:
```diff
@@ -44,20 +44,22 @@ class Model;
  * This class was introduced in OpenSim version 4.0, and enables scripting
  * (Python/MATLAB) and C++ users to postprocess their results with greater ease
- * and more versatility than with an Analysis.
+ * and flexibly than with an Analysis.
```
I suggested "flexibility" (with greater ease and [greater] flexibility than…); I don't think "flexibly" makes sense (with… flexibly than…).

chrisdembia replied:
Oops; I didn't read carefully. I have resolved this now.

tkuchida commented:
Looks good. Merging by [Rule 5](https://github.com/opensim-org/opensim-core/blob/master/CONTRIBUTING.md#making-a-pull-request-pr).

chrisdembia commented:
Thank you.
